### ISSUE TITLE
(Some but not all countries) Program metrics different between country page and program page

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#1402]
This is a release blocker because it appear to be a global issue, affecting the accurate display of metrics.

On Dev2, I'm seeing a discrepancy between program metrics displayed on the Country page and the Program page. Here is an example with a program in Afghanistan.

**Country page**
![image](https://user-images.githubusercontent.com/33670923/56152977-02396a00-5f6a-11e9-8867-606d65acbc61.png)

**Program page**
![image](https://user-images.githubusercontent.com/33670923/56152998-0a91a500-5f6a-11e9-9618-d3e75989d3d8.png)

Here's a program in Haiti

**Country page**
![image](https://user-images.githubusercontent.com/33670923/56153046-33b23580-5f6a-11e9-99ac-8f1d5c04613d.png)

**Program page**
![image](https://user-images.githubusercontent.com/33670923/56153103-3dd43400-5f6a-11e9-8682-1ac69d53e642.png)

**Comment by jennyhavah** [comment]
@camgnostic I poked around a bit more and it's _not_ a global issue. For example, I checked a few programs in Afghanistan and Tolaland that were consistent.

**Comment by RickieT** [comment]
**QA - FAILED**
* Though I did find in the reported in this ticket (afghanistan) programs matched 100% between home/program page, I am still seeing a slight difference like this one in Haiti:

![image](https://user-images.githubusercontent.com/23220180/56304009-fb3d6380-60f1-11e9-8da7-6328548ce193.png)

**Comment by camgnostic** [comment]
Update: this issue revolves around an edge case: we have an indicator in this program for which all targets are 0.  That means that while we can calculate a target (0) for the program to date, and an actual value (in this case, 14) for the program to date, we can't calculate the % over/under to determine if it's on track.  So it falls through the cracks, where it doesn't get counted as unavailable (it's got at least one completed target period and has results) and it doesn't get counted as over/under/on-target, so it doesn't get counted at all.  @jennyhavah is going to make the call on where to put these indicators on the graph, then I'll get this ticket closed out.

**Comment by RickieT** [comment]
@camgnostic so whats the resolution on this, since this was moved back the dev2 environment has been updated and im still seeing this minor difference

**Comment by jennyhavah** [comment]
@camgnostic  and I came to the resolution that in the very edgy edge case where the sum of periodic targets = 0, the indicator is treated as "unavailable."

As to why you're still seeing the discrepancy on Dev2, not sure. Cameron?

**Comment by RickieT** [comment]
@jennyhavah @camgnostic Perhaps the test environment was not up to date as expected at the time of test, but checking this morning - it looks to be this has been resolved..
![image](https://user-images.githubusercontent.com/23220180/56374475-1a9ec400-61b8-11e9-8d22-eff17d30a676.png)

**Comment by jennyhavah** [comment]
@camgnostic What's the best way for us to validate that this is fixed across all programs? 

I know one way to reproduce the problem is to modify an indicator with results, so that the targets are all zero.

Is there something else we can check?

**Comment by RickieT** [comment]
@jennyhavah the haiti program in production is having the same issue, as we do in dev2, just an fyi

**Comment by camgnostic** [comment]
For reference - there are three issues here:
1) Haiti Ann Viv Ansam program: if all your targets are zero the system was having trouble categorizing the indicator as "reporting" or "on target" or what.  We fixed that (it shows up in the "unavailable" category now)
2) Afghanistan program: Papaya-related soft-delete was breaking query logic for generating the home page data rollups, resulting in it counting a ton of deleted indicators when working out the percentage complete/targets-assigned/results-assigned/evidence-assigned values, so it was artificially deflating those numbers.  This is now fixed.
3) the 12/13 % differential.  Front end calculations were working out the number of indicators withOUT targets, backend was working out the number WITH targets, which usually results in the same percentage, but in the case of 2 indicators missing targets out of 16 indicators, you get 2/16 = 12.5, which rounds up to 13% (and then 100-13 means 87% with targets) or by the other logic you get 14/16 = 87.5 which rounds up to 88% (and then 100-88 means 12% without targets).  This is also fixed now.
